### DISCUSSION TITLE
python generator request - return values

### DISCUSSION THREAD

**Discussion by benji-finster** [langfuse/langfuse#3862] [Ideas]
### Describe the feature or potential improvement

Hi!

I noticed that your generator wrapping doesn't support return values. Was wondering if it could be added in?
Something like:

```
class Generator:
    def __init__(self, gen):
        self.gen = gen

    def __iter__(self):
        self.value = yield from self.gen
        return self.value
```

Then in `_wrap_sync_generator_result` in the try block you could do:

```
            generator = Generator(generator)
            for item in generator:
                items.append(item)

                yield item
            return generator.val
```
I want to get the return value of a generator (e.g. like in <https://stackoverflow.com/questions/34073370/best-way-to-receive-the-return-value-from-a-python-generator>), however I think currently the langfuse observe decorator destroys this.

Cheers! I hope this might be a very small but nice change.

**Comment by hassiebp** [comment]
Hi @benji-finster - thank for raising this discussion! Although this is a somewhat odd pattern as described in that [stackoverflow answer](https://stackoverflow.com/a/34073559), the decorator should not impede the original functionality of the wrapped generator. Happy to review your contribution on this if you rely on this! 🙏🏾
